src/huggingface_hub/hf_api.py

```diff
@@ -1764,10 +1893,17 @@ def upload_file(
                 The git revision to commit from. Defaults to the head of the
                 `"main"` branch.
             identical_ok (`bool`, *optional*, defaults to `True`):
-                When set to false, will raise an [HTTPError](
-                https://2.python-requests.org/en/master/api/#requests.HTTPError)
+                (Deprecated, will be removed in v0.3.0) When set to false, will
+                raise an [HTTPError](https://requests.readthedocs.io/en/latest/api/#requests.HTTPError)
                 when the file you're trying to upload already exists on the hub
                 and its content did not change.
```

Comment:
@LysandreJik @adrinjalali lmk if there's anything more I should do

Reply:
I will add that it will be removed in v0.10.0 rather than v0.3.0 (we'll be releasing v0.8.0 next week)